src/Agent.Worker/Release/ReleaseDirectoryManager.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Agent.Worker.Common;
+using Newtonsoft.Json;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
+{
+    public sealed class ReleaseDirectoryManager : AgentService, IReleaseDirectoryManager
+    {
+        public ReleaseDefinitionToFolderMap PrepareArtifactsDirectory(
```

Comment:
PrepareParentWorkingDirectory()? I thought we will have _work\1\a as ArtifactsDirectory?